app/codes/statemanager.py

```diff
@@ -0,0 +1,238 @@
+    if isinstance(wallet, str):
+        wallet = json.loads(wallet)
+    linkedstatus = wallet['specific_data']['linked_wallet'] if 'linked_wallet' in wallet['specific_data'] else False
+    if linkedstatus:
+        pid_cursor = cur.execute('SELECT person_id FROM person_wallet WHERE wallet_id=?',
+                                 (wallet['specific_data']['parentaddress'], )).fetchone()
+        pid = pid_cursor[0]
+        if not pid:
+            print("No personid linked to the parentwallet.")
+            return False
+    else:  # not a linked wallet, so create a new pid and update person table
+        hs = hashlib.blake2b(digest_size=20)
+        hs.update((wallet['wallet_address']).encode())
+        pid = 'pi' + hs.hexdigest()
+        query_params = (pid, time.mktime(datetime.datetime.now().timetuple()))
+        cur.execute(f'''INSERT OR IGNORE INTO person
+                    (person_id, created_time)
+                    VALUES (?, ?)''', query_params)
+
+    # for both new and linked wallet, update the wallet table and person_wallet table
+    kyc_doc_json = json.dumps(wallet['kyc_docs'])
+    data_json = json.dumps(wallet['specific_data'])
+    query_params = (wallet['wallet_address'],
+                    wallet['wallet_public'],
+                    wallet['custodian_wallet'],
+                    kyc_doc_json,
+                    wallet['ownertype'],
+                    wallet['jurisd'],
+                    data_json
+                    )
+    cur.execute(f'''INSERT OR IGNORE INTO wallets
+            (wallet_address, wallet_public, custodian_wallet, kyc_docs, owner_type, jurisdiction, specific_data)
+            VALUES (?, ?, ?, ?, ?, ?, ?)''', query_params)
+
+    query_params = (pid, wallet['wallet_address'])
+    cur.execute(f'''INSERT OR IGNORE INTO person_wallet
+                (person_id, wallet_id)
+                VALUES (?, ?)''', query_params)
+
+
+def add_token(cur, token, txcode=None):
+    tcodenewflag = False
+    existingflag = False
+    if 'tokencode' in token:  # creating more of an existing token or tokencode provided by user
+        if token['tokencode'] and token['tokencode'] != "0" and token['tokencode'] != "":
+            tid = cur.execute(
+                'SELECT tokencode FROM tokens WHERE tokencode=?', (token['tokencode'], )).fetchone()
+            if tid:  # tokencode exists, more of an existing token is being added to the first_owner
+                tid = tid[0]
+                existingflag = True
+            else:
+                # if provided code does not exist, it is considered new token addition
+                tid = str(token['tokencode'])
+                existingflag = False
+        else:   # mistakenly entered tokencode value as "" or "0" or 0
+            tcodenewflag = True
+            existingflag = False
+    if 'tokencode' not in token or tcodenewflag:   # new tokencode needs to be created
+        hs = hashlib.blake2b(digest_size=20)
```

Comment:
Add a uuid creator here if txcode is None
if not txcode:
    txcode = uuid.uuid()